filebeat/input/v2/input.go

```diff
@@ -0,0 +1,113 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package v2
+
+import (
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+
+	"github.com/elastic/go-concert/unison"
+)
+
+// InputManager creates and maintains actions and background processes for an
+// input type.
+// The InputManager is used to create inputs. The InputManager can provide
+// additional functionality like coordination between input of the same type,
+// custom functionality for querying or caching shared information, application
+// of common settings not unique to a particular input type, or require a more
+// specific Input interface to be implemented by the actual input.
+type InputManager interface {
+	// Init signals to InputManager to initialize internal resources.
+	// The mode tells the input manager if the Beat is actually running the inputs or
+	// if inputs are only configured for testing/validation purposes.
+	Init(grp unison.Group, mode Mode) error
+
+	// Creates builds a new Input instance from the given configuation, or returns
+	// an error if the configuation is invalid.
+	// The input must establish any connection for data collection yet. The Beat
+	// will use the Test/Run methods of the input.
+	Create(*common.Config) (Input, error)
+}
+
+// Mode tells the InputManager in which mode it is initialized.
+type Mode uint8
+
+//go:generate stringer -type Mode -trimprefix Mode
+const (
+	ModeRun Mode = iota
+	ModeTest
+	ModeOther
+)
+
+// Input is a configured input object that can be used to test or start
+// the actual data collection.
+type Input interface {
+	// Name reports the input name.
+	//
+	// XXX: check if/how we can remove this method. Currently it is required for
+	// compatibility reasons with existing interfaces in libbeat, autodiscovery
+	// and filebeat.
+	Name() string
```

Comment:
Yeah, it's not really used within the Beat itself. But it is still in here for now because the Runner interface required by config file reloading and autodiscovery requires the implementation to have a `String() string` method. The compatibility layer is using `Name` to return the string to be printed. See godoc for Runner: https://github.com/elastic/beats/blob/master/libbeat/cfgfile/reload.go#L86 

Not sure if we really need this in the future, but for now I'd keep this in order to have some compatibility with existing functionality in libbeat.